src/main/java/info/freeit/boarderpicker/service/Impl/SaleServiceImpl.java

```diff
@@ -1,58 +1,69 @@
 package info.freeit.boarderpicker.service.Impl;
 
+import info.freeit.boarderpicker.dto.BPUserDetails;
+import info.freeit.boarderpicker.dto.SaleDto;
+import info.freeit.boarderpicker.dto.UpdateSaleDto;
 import info.freeit.boarderpicker.entity.Game;
 import info.freeit.boarderpicker.entity.Sale;
 import info.freeit.boarderpicker.entity.User;
-import info.freeit.boarderpicker.exception.ObjectPersistenceException;
 import info.freeit.boarderpicker.repository.GameRepository;
 import info.freeit.boarderpicker.repository.SaleRepository;
 import info.freeit.boarderpicker.repository.UserRepository;
 import info.freeit.boarderpicker.service.SaleService;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 import java.util.List;
 @Service
+@Transactional
```

Comment:
RequiredArgsConstructor and remove hardcoded constructor

Reply:
Done